lib/ukcpio/cpio.c

```diff
@@ -58,6 +58,37 @@ int uk_syscall_r_utime(const char *, const struct utimbuf *);
 int uk_syscall_r_mkdir(const char *, mode_t);
 int uk_syscall_r_symlink(const char *, const char *);
 int uk_syscall_r_stat(const char *, struct stat *);
+int uk_syscall_r_unlinkat(int, const char *, int);
+int uk_syscall_r_rename(const char *, const char *);
+
+static int
+try_remove(const char *path)
+{
+	int r;
+
+	uk_pr_info("Trying to unlink %s\n", path);
+	r = uk_syscall_r_unlinkat(AT_FDCWD, path, 0);
+	if (!r || r == -ENOENT)
+		return 0;
+
+	uk_pr_info("Trying to rmdir %s\n", path);
+	r = uk_syscall_r_unlinkat(AT_FDCWD, path, AT_REMOVEDIR);
+	if (!r || r == -ENOENT) {
+		return 0;
+	} else {
+		char newpath[PATH_MAX];
+		char *newend = newpath + strlcpy(newpath, path, PATH_MAX);
+
+		if (newend - newpath + 2 > PATH_MAX) {
+			uk_pr_err("Cannot rename %s, path too long\n", path);
+			return -ENAMETOOLONG;
+		}
+		strcpy(newend, ".0");
+		uk_pr_info("Trying to rename %s to %s\n", path, newpath);
```

Comment:
This print will probably have more value after the function `rename` call so that we can also print the return code, I know that the caller will also do this, but I think increased verbosity here helps since RAMFS has lots of messages. I think it goes the same for the other "Trying" messages above. Since they are debug messages, a couple of additional if/else here and there for them should be fine as they would be optimized out in a performance optimized image.
E.g. `uk_pr_info("Rename operation %s to %s: %d\n", path, newpath, rc);`